plugins/pulp_rpm/plugins/distributors/iso_rsync_distributor/publish.py

```diff
@@ -0,0 +1,103 @@
+import os
+
+from pulp.plugins.rsync.publish import Publisher, RSyncPublishStep
+from pulp.plugins.util.publish_step import RSyncFastForwardUnitPublishStep
+from pulp.server.db.model import Distributor
+from pulp.server.exceptions import PulpCodedException
+
+from pulp_rpm.common import constants, ids
+from pulp_rpm.plugins import error_codes
+
+class ISORsyncPublisher(Publisher):
+
+    UNIT_FIELDS = ["_storage_path", "name"]
+
+    def _get_predistributor(self):
+        """
+        Returns the distributor that is configured as predistributor.
+        """
+        predistributor = self.get_config().flatten().get("predistributor_id", None)
```

Comment:
This may be better to not check for the value first, but instead try to use it. It would become more readable for sure.

http://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html